### PR TITLE
fix(collapse): avoid initial animation

### PR DESCRIPTION
**Pull request by bogdanalexe90**
- Ensure animation does not occur when first loaded

Fixes: https://github.com/angular-ui/bootstrap/issues/5192
Fixes: https://github.com/angular-ui/bootstrap/issues/4414

### CODE REVIEW

**Comment by Foxandxss** [comment]
Can you create a plunker with the old broken behavior and another plunker on how it end with your fix?

The test stuff is really nice (that cleanup always is handy) but this would need another test if we end deciding in merging this.

Thank you!

**Comment by bogdanalexe90** [comment]
I was able to reproduce this only with accordion plugin. For this reason i don't have any idea how to test the fix. Any suggestion are welcomed.

Before: http://plnkr.co/edit/l0Qc5rWfgCNhx4hU4JZx?p=preview
After: http://plnkr.co/edit/bp5PlTC5g6jjWDoLcleR?p=preview

**Comment by Foxandxss** [comment]
What should I do / see there? I don't see any difference when loading.

**Comment by bogdanalexe90** [comment]
ahh it was a false result. i will try to include some 3rd party libs. I'm experiencing a flickering a lot when using `ui-select` inside. Also i will try to have a look if i have some custom styles in my app.

I will try to create another plnkr soon.

**Comment by bogdanalexe90** [comment]
I identified the following problems:

1. An fixed height is set to the element
```
$animateCss(element, {
              addClass: 'in',
              easing: 'ease',
              to: { height: element[0].scrollHeight + 'px' }
            }).start()['finally'](expandDone);
```
2. Some content arrives and `element[0].scrollHeight` changes

3. The height is set to auto and the rest of the content is showed
```
element.removeClass('collapsing')
            .addClass('collapse')
            .css({height: 'auto'});
```


The problem (flickering) is only visible when you have some complex styles to render. I created a plnkr and you can see the describe behaviour with some breakpoints (on line 31 and line 44).

http://plnkr.co/edit/7dyZ3InIQ3gg69SmVAgT?p=preview

**Comment by bogdanalexe90** [comment]
I had a look on https://github.com/angular-ui/bootstrap/issues/4414 . It seems it's the same problem, i created a plunker demonstrating the fix:

http://plnkr.co/edit/Y8Uf0KKDq2ubMxAa71dg?p=preview

**Comment by wesleycho** [comment]
This needs a test still in order for us to accept it.

File: src/collapse/test/collapse.spec.js [hunk 1]
```diff
@@ -1,55 +1,76 @@
-describe('collapse directive', function() {
+describe('collapse directive', function () {
```
Comment (wesleycho) [inline src/collapse/test/collapse.spec.js]: This whitespace change is unnecessary - this will need to all be changed back on merge.

**Comment by wesleycho** [comment]
Minus all of the whitespace changes that should be undone, what are your thoughts on this @Foxandxss ?

**Comment by wesleycho** [comment]
Can you squash your commits into 1?

**Pull request closed without merging**